tests/suites/test_suite_pkparse.data

```diff
@@ -1197,8 +1197,8 @@ depends_on:MBEDTLS_PK_HAVE_ECC_KEYS
 pk_parse_key:"30070201010400a000":MBEDTLS_ERR_PK_KEY_INVALID_FORMAT
 
 Key ASN1 (OneAsymmetricKey X25519, doesn't match masking requirements, from RFC8410 Appendix A but made into version 0)
-depends_on:MBEDTLS_ECP_C
-pk_parse_key:"302e020100300506032b656e04220420f8ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff3f":MBEDTLS_ERR_PK_KEY_INVALID_FORMAT
+depends_on:MBEDTLS_PK_HAVE_ECC_KEYS
+pk_parse_key:"302e020100300506032b656e04220420f8ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff3f":0
```

Comment:
I agree with your request, but having the code to always do `mbedtls_ecp_check_privkey` on the imported keys isn't enough to guarantee that the key is OK?

Reply:
Since the call to `mbedtls_ecp_check_privkey()` is part of `mbedtls_ecp_read_key()`, it seems to me that if someone accidentally changed `mbedtls_ecp_read_key()` back to `mbedtls_mpi_read_binary_le()` (without adding back a manual call to `mbedtls_ecp_check_privkey()` of course), then this test would still pass.

I would like a test that would catch this kind of possible future mistake. If I'm mistaken and the present one does, great (but please explain what I missed)! Otherwise, let's extend it.

Reply:
Oh OK, now I get it: it's something that we put in place to prevent some erroneous change in the future. Ok, I'll work on this ;)

Reply:
Done in [this commit](https://github.com/Mbed-TLS/mbedtls/pull/7863/commits/e4bad0dec6a571fa3a29853e1cee149c7bc40cac).
However instead of modifying all the already existing test cases, since this issue/feature seems to me to be very specific of Montgomery keys, I preferred to add a specific function and test for it. Right now it only involves x25519, but if you prefer we can extend it to x448 (though in this case I need to find a valid key to test with ':))